instancio-core/src/main/java/org/instancio/internal/generator/domain/id/bra/CpfGenerator.java

```diff
@@ -0,0 +1,97 @@
+package org.instancio.internal.generator.domain.id.bra;
+
+import org.instancio.Random;
+import org.instancio.generator.GeneratorContext;
+import org.instancio.generator.specs.bra.CpfSpec;
+import org.instancio.internal.generator.AbstractGenerator;
+
+public class CpfGenerator extends AbstractGenerator<String> implements CpfSpec {
+
+    private static final int CPF_LENGTH = 9;
+
+    private boolean format;
+
+    public CpfGenerator(final GeneratorContext context){
+        super(context);
+    }
+
+    @Override
+    public CpfGenerator format(Boolean format) {
+        this.format = format;
+        return this;
+    }
+
+    @Override
+    public String apiMethod() {
+        return "cpf()";
+    }
+
+    @Override
+    public CpfGenerator nullable() {
+        super.nullable();
+        return this;
+    }
+
+    @Override
+    protected String tryGenerateNonNull(Random random) {
+        String baseCpf = random.digits(CPF_LENGTH);
+        int firstDigitVerifier = generateFirstDigit(baseCpf);
+        int secondDigitVerifier = generateSecondDigit(baseCpf, firstDigitVerifier);
+        if(!format){
+            return baseCpf+firstDigitVerifier+secondDigitVerifier;
+        }
+        return format(baseCpf, String.format("%d%d", firstDigitVerifier, secondDigitVerifier));
+    }
+
+    /**
+     * Formats the given CPF using the following pattern: @code{xxx.xxx.xxx-xx}
+     * @param baseCPF The base CPF
+     * @param digits The verifier digits
+     * @return The CPF formatted
+     */
+    String format(String baseCPF, String digits) {
+        StringBuilder result = new StringBuilder();
+        result.append(baseCPF, 0, 3);
+        result.append(".");
+        result.append(baseCPF, 3, 6);
+        result.append(".");
+        result.append(baseCPF, 6, 9);
+        result.append("-");
+        result.append(digits);
+        return result.toString();
+    }
+
+    /**
+     * Generates the first CPF digit verifier
+     *
+     * @param baseCpf The base random CPF
+     * @return the first CPF verifier
+     */
+    int generateFirstDigit(String baseCpf) {
```

Comment:
I think we can make this method `private` and remove the unit tests for this method, since we already have Hibernate validator tests that verify the generated values are correct.

Reply:
Okay! Makes sense! I added the unit tests before using the Hibernate Validator! I removed those tests now!